pkg/controllers/node/node_controller.go

```diff
@@ -0,0 +1,111 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package node
+
+import (
+	"context"
+	"os"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/metrics"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apimachineryruntime "k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
+	servicecommon "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/node"
+)
+
+var (
+	log               = logger.Log
+	MetricResTypeNode = common.MetricResTypeNode
+)
+
+// NodeReconciler reconciles a Node object
+type NodeReconciler struct {
+	client.Client
+	Scheme  *apimachineryruntime.Scheme
+	Service *node.NodeService
+}
+
+func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	node := &v1.Node{}
+	deleted := false
+	log.Info("reconciling node", "node", req.NamespacedName)
+
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, MetricResTypeNode)
+
+	if err := r.Client.Get(ctx, req.NamespacedName, node); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("node not found, may be deleted", "req", req.NamespacedName)
```

Comment:
`may be deleted`: let's remove this assumption

Reply:
done